src/core/config/border_agent.h

```diff
@@ -72,7 +72,7 @@
  *
  */
 #ifndef OPENTHREAD_CONFIG_BORDER_AGENT_ID_ENABLE
-#define OPENTHREAD_CONFIG_BORDER_AGENT_ID_ENABLE 0
+#define OPENTHREAD_CONFIG_BORDER_AGENT_ID_ENABLE OPENTHREAD_CONFIG_BORDER_AGENT_ENABLE
```

Comment:
From https://github.com/openthread/openthread/pull/9794#pullrequestreview-1839756667:

> This works, but we can also define default to be `1` (since  `OPENTHREAD_CONFIG_BORDER_AGENT_ID_ENABLE` would only be used when we have `BORDER_AGENT_ENABLE`?)

Thoughts?

Reply:
Thanks raising that again. I updated the code to make it more consistent with other features. 

Seems the convention is, defining the child feature configuration as a literal assuming the parent feature is enabled.